frame/system/src/lib.rs

```diff
@@ -953,6 +953,28 @@ impl<
 	}
 }
 
+/// The "AND gate" implementation of `EnsureOrigin`.
+///
+/// Origin check will pass if `L` and `R` origin check passes. `L` is tested first.
+pub struct EnsureBothOf<AccountId, L, R>(sp_std::marker::PhantomData<(AccountId, L, R)>);
+impl<
+		AccountId,
+		O: Into<Result<RawOrigin<AccountId>, O>> + From<RawOrigin<AccountId>> + Clone,
+		L: EnsureOrigin<O>,
+		R: EnsureOrigin<O>,
+	> EnsureOrigin<O> for EnsureBothOf<AccountId, L, R>
+{
+	type Success = (L::Success, R::Success);
+	fn try_origin(o: O) -> Result<Self::Success, O> {
+		L::try_origin(o.clone()).map_or_else(|o| Err(o), |l| R::try_origin(o).map(|r| (l, r)))
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn successful_origin() -> O {
+		L::successful_origin()
```

Comment:
This is "wrong". Because a successful origin of `L` doesn't mean it will be successful for `R`.

Reply:
Indeed, I see no easy implementation, maybe we can start creating a origin variant only when the feature runtime-benchmarks is activated.
Like frame_system would define a new variant `AlwaysSuccess`.
But I don't very like this solution.

Reply:
Yea, have missed this case. Updating implementation.